main.py

```diff
@@ -0,0 +1,237 @@
+from datetime import datetime
+from constants import RED, BLUE, RESET, months 
+
+
+"""_summary_
+    LoadFileData class loads the data from files and make data structure of the data which coming from files
+    [
+        dict of every single file
+        {
+            dict of every single date with their fields and value
+            date: {
+                fields:  value
+            }    
+        }
+    ]
+"""
+class LoadFileData:
+    def __init__(self, year, month):
+        self.year = year
+        self.month = month
+        self.data = []
+        self.readDatadromFiles()
+        
+    """_summary_
+        read data from every single file 
+        if month is None then it iterate over all the months that is present in the months global variable
+        if month is present then it only iterate over the single month
+    """
+    def readDatadromFiles(self):
+        if self.month == None:
+            for month in months.values():
+                self.readDataFromSingleFile(month)
+                
+            if len(self.data) == 0:
+                raise Exception("No file is found")
+        else: 
+            self.readDataFromSingleFile(months[self.month])
+            if len(self.data) == 0:
+                raise Exception("No file is found")
```

Comment:
https://github.com/developer-jawad/Weatherman/pull/1/files#r1615940065

We have code duplication, if we keep it inside if/else. Thought?